src/ensembl/src/content/app/browser/track-panel/track-panel-list/TrackPanelListItem.test.tsx

```diff
@@ -26,92 +30,177 @@ import {
 import { Status } from 'src/shared/types/status';
+import { DrawerView } from 'src/content/app/browser/drawer/drawerState';
 
-describe('<TrackPanelListItem />', () => {
-  beforeEach(() => {
-    jest.resetAllMocks();
-  });
+import * as drawerActions from '../../drawer/drawerActions';
+import * as browserActions from 'src/content/app/browser/browserActions';
+import * as trackPanelActions from 'src/content/app/browser/track-panel/trackPanelActions';
+import { isTrackCollapsed } from 'src/content/app/browser/track-panel/trackPanelSelectors';
+
+jest.mock('../../drawer/drawerActions', () => {
+  const originalModule = jest.requireActual('../../drawer/drawerActions');
 
-  const defaultProps: TrackPanelListItemProps = {
-    categoryName: faker.lorem.words(),
-    trackStatus: Status.SELECTED,
-    defaultTrackStatus: Status.SELECTED,
-    track: createMainTrackInfo(),
-    activeGenomeId: faker.lorem.words(),
-    activeEnsObjectId: faker.lorem.words(),
-    isDrawerOpened: false,
-    drawerView: 'bookmarks',
-    highlightedTrackId: faker.lorem.words(),
-    isCollapsed: false,
+  return {
+    ...originalModule,
     changeDrawerView: jest.fn(),
-    toggleDrawer: jest.fn(),
-    updateTrackStatesAndSave: jest.fn(),
+    setActiveDrawerTrackId: jest.fn()
+  };
+});
+
+jest.mock('src/content/app/browser/browserActions', () => {
+  const originalModule = jest.requireActual(
+    'src/content/app/browser/browserActions'
+  );
+
+  return {
+    ...originalModule,
+    updateTrackStatesAndSave: jest.fn()
+  };
+});
+
+jest.mock('src/content/app/browser/track-panel/trackPanelActions', () => {
+  const originalModule = jest.requireActual(
+    'src/content/app/browser/track-panel/trackPanelActions'
+  );
+
+  return {
+    ...originalModule,
     updateCollapsedTrackIds: jest.fn()
   };
+});
+
+const fakeGenomeId = faker.lorem.words();
+
+const mockState = {
+  drawer: {
+    isDrawerOpened: { [fakeGenomeId]: true },
+    drawerView: { [fakeGenomeId]: DrawerView.BOOKMARKS },
+    activeDrawerTrackIds: {}
+  },
+  browser: {
+    browserEntity: {
+      activeGenomeId: fakeGenomeId,
+      activeEnsObjectIds: {
+        [fakeGenomeId]: faker.lorem.words()
+      }
+    },
+    trackPanel: {
+      [fakeGenomeId]: {
+        highlightedTrackId: faker.lorem.words(),
+        collapsedTrackIds: []
+      }
+    }
+  }
+};
+
+jest.mock('react-redux', () => {
+  const originalModule = jest.requireActual('react-redux');
+  return {
+    ...originalModule,
+    useDispatch: () => jest.fn(),
+    useSelector: (selector: any) => {
+      return selector(mockState);
+    }
+  };
+});
+
+const mockStore = configureMockStore([thunk]);
+
+const defaultProps: TrackPanelListItemProps = {
+  categoryName: faker.lorem.words(),
+  trackStatus: Status.SELECTED,
+  defaultTrackStatus: Status.SELECTED,
+  track: createMainTrackInfo()
+};
+
+const wrapInRedux = (
+  props?: Partial<TrackPanelListItemProps>,
+  state: typeof mockState = mockState
```

Comment:
1) I don't think you are passing custom props to this wrapper in any of the tests
2) If we want to keep the option of passing the props; perhaps consider a single object with the `state` and the `props` fields as a parameter of `wrapInRedux`. Passing `undefined` as a first parameter to `wrapInRedux` is awkward.

Reply:
I've removed the param `props` for now.